### ISSUE TITLE
@RequiresNonNull and field visibilty

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#391]
```
Hi,

from what I gather @RequiresNonNull should be enforced by the nullness checker 
but I can't get it to work.

I have this (condensed) test case that produces no errors for these classes, 
only warnings about missing documentation:

class ClassA {
    @Nullable private String value = null;

    @RequiresNonNull("value") public String getValue() {
        return value;
    }
}

public class Tester {
    public static void main(String[] args) {
        ClassA a = new ClassA();
        a.getValue();
    }
}

No error is reported even with
@MonotonicNonNull private String value;
which is what I had in the beginning, but then I tried and noticed that it 
doesn't work even with the code above that explicitly sets the field to null.

I'm using version 1.8.9 with NetBeans 8.0.2 x64, Ant 1.9.4 and the javac 
included in JDK 1.8.0_u25 x64 on Windows 8.
I pass these options to the compiler:
-version -Xlint:all -Xdoclint:all

My ant target is attached.

The nullness checker seems to work fine otherwise, I can see various errors for 
other classes that seem sensible and I already have added some annotations to 
eliminate errors.
So I think that the issue, if there's any, is related to the particular 
annotations used in the sample above.
```

Original issue reported on code.google.com by `karlsand...@gmail.com` on 11 Jan 2015 at 9:22

Attachments:
- [Ant Task.xml](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-391/comment-0/Ant Task.xml)

**Comment by jthaine** [comment]
This issue has been fixed for private fields (but not yet for private methods) via commit 85e85ab1719ebfc0b3fa558ac7e08dd0f07f5b9e

**Comment by mernst** [comment]
Because this is fixed for private fields, I am closing the issue.  If private methods become important, please file a new issue.
